vllm/model_executor/utils.py

```diff
@@ -28,4 +29,22 @@ def set_weight_attrs(
     for key, value in weight_attrs.items():
         assert not hasattr(
             weight, key), (f"Overwriting existing tensor attribute: {key}")
+
+        # NOTE(woosuk): For TPU, param.data.copy_(weight) happens lazily,
```

Comment:
to be more accurate this is because in VLLM we do
```
narrowed_tensor = param.data.narrow(0, offset, len)
narrowed_tensor.copy_(real_weight)
```
`narrowed_tensor` and `param.data` share the same storage. With functionization, the in place update on the `narrowed_tensor` will lazily propagate to the base tensor which is `param.data`.

Reply:
Thanks for the elaboration. Fixed the comment!